arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,360 @@
+    /// The network attributes for the comparison.
+    var phasesNetworkAttribute: AGSUtilityNetworkAttribute!
+    /// A list of possible phases populated from the network's attributes.
+    var excludedPhases = [AGSCodedValue]()
+    /// A list of phases that are included in the load report.
+    var includedPhases = [AGSCodedValue]() {
+        didSet {
+            let isEmpty = includedPhases.isEmpty
+            DispatchQueue.main.async { [weak self] in
+                self?.runBarButtonItem.isEnabled = !isEmpty
+            }
+        }
+    }
+    /// The phase summaries in the load report.
+    var summaries = [AGSCodedValue: PhaseSummary]() {
+        didSet {
+            if !resetBarButtonItem.isEnabled {
+                DispatchQueue.main.async { [weak self] in
+                    self?.resetBarButtonItem.isEnabled = true
+                }
+            }
+        }
+    }
+    
+    /// The number formatter for phase summaries.
+    let numberFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        formatter.usesGroupingSeparator = true
+        return formatter
+    }()
+    
+    /// A struct for the phase summary, which contains the total customers
+    /// and total load for the phase.
+    struct PhaseSummary {
+        let totalCustomers: Int
+        let totalLoad: Int
+    }
+    
+    // MARK: Methods
+    
+    /// Load the utility network.
+    func loadUtilityNetwork() {
+        SVProgressHUD.show(withStatus: "Loading utility network…")
+        utilityNetwork.load { [weak self] error in
+            SVProgressHUD.dismiss()
+            guard let self = self else { return }
+            guard error == nil else {
+                self.presentAlert(error: error!)
+                return
+            }
+            // Create a default starting location.
+            guard let startingLocation = self.makeStartingLocation() else {
+                self.presentAlert(message: "Fail to create starting location.")
+                return
+            }
+            // Get the base condition and trace configuration from a default tier.
+            guard let utilityTierConfiguration = self.getTraceConfiguration() else {
+                self.presentAlert(message: "Fail to get trace configuration.")
+                return
+            }
+            // Proceed if the utility network loaded without issue.
+            self.utilityNetworkDidLoad(startingLocation: startingLocation, traceConfiguration: utilityTierConfiguration)
+        }
+    }
+    
+    /// Called in response to the utility network load operation completing.
+    /// - Parameters:
+    ///   - startingLocation: The utility element to start the trace from.
+    ///   - traceConfiguration: The utility tier's trace configuration.
+    func utilityNetworkDidLoad(startingLocation: AGSUtilityElement, traceConfiguration: AGSUtilityTraceConfiguration) {
```

Comment:
After housekeeping the old monstrously lengthy method, I'm still not quite satisfied with this one. There are a few things going on after the utility network is loaded

- Get a list of phases or `[AGSCodedValue]`
- Create a base `AGSUtilityTraceParameters` with some default configurations
- Hold reference to a few properties

The difficulty is that getting the trace parameters depends on multiple preliminary steps and unwrapped variables.  Looking for more suggestions.